### PR TITLE
Added 2014-12-28-Rtree-example.Rmd

### PR DESCRIPTION
**Pull request by teramonagi**
Hi Mr.Dirk; 

I really appreciate your kindness in
- https://github.com/eddelbuettel/bh/issues/5

I went straight to make a example using this.

### CODE REVIEW

File: src/2014-12-28-Rtree-example.Rmd [hunk 1]
```diff
@@ -0,0 +1,140 @@
+---
+title: Create R-tree data structure using Rcpp and BH package(Boost::Geometry).
+author: teramonagi
```
Comment (eddelbuettel) [inline src/2014-12-28-Rtree-example.Rmd:3]: Document looks good, but may I ask you to follow custom and sign with something like a firstname lastname?

**Comment by eddelbuettel** [comment]
Thanks so much for the submission -- it looks very promising so far (though the one comment regarding the header).  I'll work on it later today.

**Comment by eddelbuettel** [comment]
I had to change it from .Rmd into .cpp so that we have one compilation unit in which the Module is visible.

**Comment by eddelbuettel** [comment]
@jjallaire @olmjo need help -- can we do Modules?  I can't think of an existing example.

The file `sourceCpp()`'s just fine, but that is now what knitr does, is it?  If the chunks get processed piece by piece then this submission may not work, sadly.

**Comment by jjallaire** [comment]
Yes we can in fact do modules (the conversion you made to cpp is indeed the
only way to get the Rcpp bits processed as a single chunk)

**Comment by eddelbuettel** [comment]
Shucks. I just converted to standard function. Hm. I'll just commit what I have now--as well as the 'predecessor' which did not work for me.  The final R chunk seemed not to know about the preceding code.

**Comment by eddelbuettel** [comment]
All yours -- if you can spare a moment.  Otherwise a self-sufficient (but cruder) post should be up in a moment.

**Comment by eddelbuettel** [comment]
It may just need the equivalent of the `// [[Rcpp::export]]` tag but I cannot recall what that was for Modules :-/

**Comment by teramonagi** [comment]
Hi guys;

Thank you for assisting my codes :)

**Comment by jjallaire** [comment]
For modules we just look for the RCPP_MODULE macro (no special attributes are required). We then call Rcpp::populate automatically.

**Comment by eddelbuettel** [comment]
Right. And the `RCPP_MODULE`  is there.  

But do you understand why compilation then fails with

```
Building shared library for Rcpp code chunk...
Quitting from lines 62-187 () 
Error: could not find function "new"
```

for code as simple as

``` .c++
// Rcpp modules allow programmers to expose C++ functions/classes to R easily
RCPP_MODULE(RTreeCpp){
    class_<RTreeCpp>("RTreeCpp")
        .constructor<Rcpp::DataFrame>()
        .method( "knn", &RTreeCpp::knn )
    ;
}
```

Adding `library(methods)` or other stop gap meeasure does not help either.  Call me puzzled.

**Comment by jjallaire** [comment]
I'm not sure why that fails (color me even more puzzled). Here's a simple example based on the code in the Rcpp modules vignette that works for me locally against both Rcpp CRAN and Rcpp master: https://gist.github.com/jjallaire/9e9912a109e1e12fe97b

**Comment by eddelbuettel** [comment]
I am very sure it works "per se".  The code submitted by tomoaki works _when used via sourceCpp()_.
So the code is fine, and Rcpp is fine.

Sadly, it doesn't work via `knitr.sh` in the Gallery. We drop something along the way. And that has me very confused because this works with everything else :-/

**Comment by jjallaire** [comment]
I took a few minutes to look over knit.sh and nothing pops out at me right
away as the source of the problem. I'll dig into it a bit more to see if
it's easy to sort out.

On Sun, Dec 28, 2014 at 4:25 PM, Dirk Eddelbuettel <notifications@github.com

**Pull request merged**